pkg/sandbox/runx/impl/registry/registry.go

```diff
@@ -118,7 +118,12 @@ func (r *Registry) GetPackage(ctx context.Context, ref types.PkgRef, platform ty
 		return "", err
 	}
 
-	err = Extract(ctx, artifactPath, installPath.String())
+	if isKnownArchive(filepath.Base(artifactPath)) {
+		err = Extract(ctx, artifactPath, installPath.String())
+	} else {
+		// If we can't extract, treat as binary
```

Comment:
Checking the first 4 bytes would be a pretty good heuristic for determining if it's meant to be executable:

- `!#` = shell script
- `0x7f454c46` = ELF (Linux)
- `0xfeedface` = macho 32 bit binary (macOS)
- `0xfeedfacf` = macho 64 bit binary (macOS)
- `0xcafebabe` = macho universal binary (macOS)

Reply:
Looks like I also need to add little endian for mac binaries